src/Microsoft.TemplateEngine.Cli/TemplateListResolutionResult.cs

```diff
@@ -108,7 +109,13 @@ public bool TryGetAllInvokableTemplates(out IReadOnlyList<ITemplateMatchInfo> in
 
         public bool TryGetSingularInvokableMatch(out ITemplateMatchInfo template)
         {
-            IReadOnlyList<ITemplateMatchInfo> invokableMatches = _coreMatchedTemplates.Where(x => x.IsInvokableMatch()).ToList();
+            IReadOnlyList<ITemplateMatchInfo> invokableMatches = _coreMatchedTemplates.Where(x => x.IsInvokableMatch() 
+                                                && (_hasUserInputLanguage 
+                                                    || !x.DispositionOfDefaults.Any()
+                                                    || x.DispositionOfDefaults.Any(y => y.Location == MatchLocation.DefaultLanguage && y.Kind == MatchKind.Exact))
+                                                    )
+                                                    .ToList();
+            
             if (invokableMatches.Count() == 1)
```

Comment:
Just noticed, we should change this to the property `Count` rather than the extension method `Count()`